lib/galaxy/model/tool_shed_install/__init__.py

```diff
@@ -92,6 +92,8 @@ def get_shed_config_dict(self, app, default=None):
         Return the in-memory version of the shed_tool_conf file, which is stored in the config_elems entry
         in the shed_tool_conf_dict.
         """
+        if default is None:
+            default = {}
```

Comment:
I'd just remove the `default` argument for both this method and `guess_shed_config()`, it's never used in their calls except in `lib/tool_shed/metadata/metadata_generator.py` , where it's set to `{}` anyway.

Also, I think on line 105:
```python
            self.guess_shed_config(app, default=default)
```

should be:

```python
            return self.guess_shed_config(app, default=default)
```

or, if you agree with the suggestion above:

```python
            return self.guess_shed_config(app)
```

Reply:
Yep, makes sense to me.